src/Aspire.Dashboard/Components/Controls/SelectResourceOptions.razor.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+
+namespace Aspire.Dashboard.Components;
+
+public partial class SelectResourceOptions<TValue>
+{
+    private async Task OnAllValuesCheckedChangedInternalAsync(bool? newAreAllVisible)
+    {
+        AreAllVisible = newAreAllVisible;
```

Comment:
Previously this line (and the logic in OnValueVisibilityChangedInternalAsync) was done in the callback but can be done directly in this component before invoking the callback.